src/alg/transform.rs

```diff
@@ -0,0 +1,184 @@
+use core::f64;
+use std::{
+    cell::RefCell,
+    ffi::{CStr, CString},
+};
+
+use libc::c_void;
+
+use crate::{
+    errors::{GdalError, Result},
+    Dataset,
+};
+
+/// Ground Control Point (GCP) used to georeference a dataset.
+#[derive(Debug, Clone)]
+pub struct GCP {
+    id: String,
+    info: Option<String>,
+    pixel_xy: (usize, usize),
+    location_xyz: (f64, f64, f64),
+}
+
+impl GCP {
+    /// Returns a GCP constructed from its GDAL C equivalent.
+    ///
+    /// # Arguments
+    ///
+    /// * `c_gcp` - A valid pointer to a C GDAL GCP representation.
+    ///
+    /// # Safety
+    ///
+    /// The pointer specified by `c_gcp` must point to a valid memory location.
+    pub unsafe fn with_c_gcp(c_gcp: *const gdal_sys::GDAL_GCP) -> GCP {
+        GCP {
+            id: CStr::from_ptr((*c_gcp).pszId).to_str().unwrap().to_string(),
+            info: CStr::from_ptr((*c_gcp).pszInfo)
+                .to_str()
+                .map_or(None, |str| Some(str.to_string())),
+            pixel_xy: ((*c_gcp).dfGCPPixel as usize, (*c_gcp).dfGCPLine as usize),
+            location_xyz: ((*c_gcp).dfGCPX, (*c_gcp).dfGCPY, (*c_gcp).dfGCPZ),
+        }
+    }
+}
+
+/// Polynomial order.
+#[derive(Copy, Clone)]
+pub enum Order {
+    First = 1,
+    Second = 2,
+    Third = 3,
+}
+/// Transformer used to map between pixel/line/height to longitude/latitude/height.
+pub struct Transformer {
+    c_transformer_ref: RefCell<*mut c_void>,
+}
+
+impl Transformer {
+    /// Construct a ```Transformer``` from a valid C GDAL pointer to a transformer instance.
+    pub(crate) unsafe fn with_c_transformer(c_transformer: *mut c_void) -> Transformer {
+        Transformer {
+            c_transformer_ref: RefCell::new(c_transformer),
+        }
+    }
+
+    /// Extracts the Ground Control Points from the specified dataset.
+    ///
+    /// # Arguments
+    ///
+    /// * `dataset` - The `Dataset` to extract Ground Control Points from.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use std::path::Path;
```

Comment:
Does `rustdoc` still compile this without `rust` in fence suffix. i.e. does this need to be `rust, no_run`?